src/logic/calculate.js

```diff
@@ -0,0 +1,25 @@
+import operate from './operate';
+
+function calculate(calculator, buttonName) {
+  let { total, next, operation } = calculator;
+
+  if (buttonName === '+/-') {
+    total *= -1;
+    next *= -1;
+  } else if (
+    buttonName === '+'
+    || buttonName === '-'
+    || buttonName === 'X'
+    || buttonName === '%'
+    || buttonName === '÷'
+  ) {
+    total = operate(total, next, operation);
+  } else if (buttonName === 'AC') {
```

Comment:
### Heads-Up:
**we need to add more cases;** 
1. if the user divides a number by zero we should return an error message. e.g `4/0`
2. if the `buttonName`  = `.` we should return `0.`